plugins/woocommerce/includes/legacy/class-wc-legacy-api.php

```diff
@@ -90,30 +146,72 @@ public function handle_rest_api_requests() {
 
-			wc_maybe_define_constant( 'WC_API_REQUEST', true );
-			wc_maybe_define_constant( 'WC_API_REQUEST_VERSION', absint( $wp->query_vars['wc-api-version'] ) );
+		wc_maybe_define_constant( 'WC_API_REQUEST', true );
+		wc_maybe_define_constant( 'WC_API_REQUEST_VERSION', absint( $wp->query_vars['wc-api-version'] ) );
+
+		$route =  $wp->query_vars['wc-api-route'];
+		$this->maybe_log_rest_api_request($route, $_SERVER['HTTP_USER_AGENT'] ?? null);
 
-			// Legacy v1 API request.
-			if ( 1 === WC_API_REQUEST_VERSION ) {
-				$this->handle_v1_rest_api_request();
-			} elseif ( 2 === WC_API_REQUEST_VERSION ) {
-				$this->handle_v2_rest_api_request();
-			} else {
-				$this->includes();
+		// Legacy v1 API request.
+		if ( 1 === WC_API_REQUEST_VERSION ) {
+			$this->handle_v1_rest_api_request();
+		} elseif ( 2 === WC_API_REQUEST_VERSION ) {
+			$this->handle_v2_rest_api_request();
+		} else {
+			$this->includes();
 
-				$this->server = new WC_API_Server( $wp->query_vars['wc-api-route'] );
+			$this->server = new WC_API_Server( $route );
 
-				// load API resource classes.
-				$this->register_resources( $this->server );
+			// load API resource classes.
+			$this->register_resources( $this->server );
 
-				// Fire off the request.
-				$this->server->serve_request();
+			// Fire off the request.
+			$this->server->serve_request();
+		}
+	}
+
+	/**
+	 * Display an admin notice with information about the last Legacy REST API usage,
+	 * if the corresponding transient is available and unless the Legacy REST API
+	 * extension is installed or the user has dismissed the notice.
+	 */
+	private function maybe_display_legacy_wc_api_usage_notice(): void {
+		if ( is_plugin_active( 'woocommerce-legacy-rest-api/woocommerce-legacy-rest-api.php' ) ) {
+			if ( WC_Admin_Notices::has_notice( 'legacy_api_usages_detected' ) ) {
+				WC_Admin_Notices::remove_notice( 'legacy_api_usages_detected', true );
 			}
+			return;
+		}
 
-			exit;
+		$legacy_api_usages = get_transient( 'wc_legacy_rest_api_usages' );
+		if ( false !== $legacy_api_usages && ! WC_Admin_Notices::user_has_dismissed_notice( 'legacy_api_usages_detected' ) ) {
+			unset( $legacy_api_usages['user_agents']['unknown'] );
+			$user_agents = array_keys( $legacy_api_usages['user_agents'] );
+
+			WC_Admin_Notices::add_custom_notice(
+				'legacy_api_usages_detected',
+				sprintf(
+					'%s%s',
+					sprintf(
+						'<h4>%s</h4>',
+						esc_html__( 'WooCommerce Legacy REST API access detected', 'woocommerce' )
+					),
+					sprintf(
+					// translators: Placeholders are URLs.
+						wpautop( wp_kses_data( __( '<p>The WooCommerce Legacy REST API has been accessed <b>%1$d</b> times since <b>%2$s</b>. There are <b>%3$d</b> known user agents registered. There are more details in <b><a target="_blank" href="%4$s">the WooCommerce log files</a></b> (file names start with <code>legacy_rest_api_usages</code>).', 'woocommerce' ) ) ),
```

Comment:
Given there are several numbers involved, I understand it's difficult to pass this through `_n()` properly, but we could use text like "time(s)" and "user agent(s)" to sort of indicate that the number might or might not be plural.